### ISSUE TITLE
[BUG] PDO query traces no longer show query, but "Array"

### ISSUE THREAD

**Issue by ben-challis** [DataDog/dd-trace-php#857]
After upgrading to 0.43.0, PDO queries now display "Array" as their resource name, rather than the SQL. Previous version of 0.38.0 was not impacted, currently on 0.42.0 and not impacted. We also experienced issue #855.

![image](https://user-images.githubusercontent.com/8275163/80586467-eab20f80-8a0c-11ea-81a1-70fb352f77eb.png)

**PHP Info**
- Output of `php -v`
```
PHP 7.4.4 (cli) (built: Mar 24 2020 01:34:16) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.4, Copyright (c), by Zend Technologies
    with blackfire v1.32.0~linux-musl-x64-non_zts74, https://blackfire.io, by Blackfire
```

- Output of `php -m`
```[PHP Modules]
bcmath
blackfire
Core
ctype
curl
date
decimal
dom
fileinfo
filter
ftp
hash
iconv
intl
json
libxml
mbstring
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
pdo_sqlite
Phar
posix
readline
Reflection
session
SimpleXML
sockets
sodium
SPL
sqlite3
standard
tokenizer
xml
xmlreader
xmlwriter
Zend OPcache
zlib

[Zend Modules]
Zend OPcache
blackfire
```

**OS Info**
- Alpine Linux 3.11, Kubernetes

**Comment by SammyK** [comment]
Hey @ben-challis! Thanks for reporting this issue. I'm suspecting that this might be related to the behavior change introduced in #762. It's documented under the "Argument Mutation" heading in that PR. To confirm this suspicion, can you confirm the following:

1. There is a custom implementation of PDO being used that extends directly from the base `PDO` class.
2. The custom implementation mutates the `$statement` argument for one or more of the following methods: `PDO::query()`, `PDO::exec()`, `PDO::prepare()`, and converts it to an array.

Basically, something similar to this:

```php
class MyPDO extends \PDO {
	public function query($statement) {
		$statement = [$statement/*,  .. */];
		// ...
	}
}
```

**Comment by ben-challis** [comment]
Hey @SammyK, there is indeed an extension to PDO, but it doesn't appear to mutate `$statement`, this is from `doctrine/dbal:2.10.2`. We have no further PDO extensions.

Let me know if you need any further info?


```php
// vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php

/**
 * PDO implementation of the Connection interface.
 * Used by all PDO-based drivers.
 */
class PDOConnection extends PDO implements Connection, ServerInfoAwareConnection
{
// snip

    /**
     * {@inheritdoc}
     */
    public function query()
    {
        $args = func_get_args();

        try {
            $stmt = parent::query(...$args);
            assert($stmt instanceof \PDOStatement);

            return $stmt;
        } catch (\PDOException $exception) {
            throw new PDOException($exception);
        }
    }
```

I've also confirmed this code path (`PDOConnection::query()`) executes in the spans with issues.

**Comment by SammyK** [comment]
Thanks for confirming this @ben-challis. I can confirm that in this case the `PDO::query()` function signature change is the root cause of this issue. We will work on a fix for this and I'll give you a status update next week.

Thanks again for reporting this issue!

**Comment by SammyK** [comment]
Hey @ben-challis! We have a fix for this issue in #872 and it should be included in our release this week. I'll follow up when the release is ready.

**Comment by SammyK** [comment]
Hey @ben-challis! I just wanted to give you a status update on this. We've slated 0.45.0 for release on Tuesday (5/12). I'll follow up once it is released. Have a great weekend!

**Comment by labbati** [comment]
Hi @ben-challis 0.45.0 is out that fixes this issue. I going ahead and close it, please feel free to re-open it if you still see any issue.
Thanks again for taking the time to open it and for the patience.
